src/python/pants/option/global_options.py

```diff
@@ -414,14 +414,23 @@ def register_bootstrap_options(cls, register):
             help="Verify that all config file values correspond to known options.",
         )
 
+        register(
+            "--build-patterns",
+            advanced=True,
+            type=list,
+            default=["BUILD", "BUILD.*"],
+            help="The naming scheme for BUILD files, i.e. where you define targets.",
```

Comment:
Other places where we define patterns, they tend to be gitignore, but this one is 1) name only, 2) uses a different list to ignore things. Probably worth mentioning both of those.

I sortof wonder whether it is necessary to expose the patterns, vs just exposing `ignore`, but if the two lists up looking consistent then it's fairly harmless I guess.